### ISSUE TITLE
Auto Parse FileName from Content-Disposition or URL

### ISSUE THREAD

**Issue by xwinie** [go-resty/resty#926]
希望这两个支持目录设置，不需要指定文件名设置

**Comment by jeevatkm** [comment]
Translated to English with Google -
```
From
希望这两个支持目录设置，不需要指定文件名设置

To
Hopefully these two will support directory settings without having to specify a file name
```

I need help understanding your question/ask. Can you provide additional details?

**Comment by RA341** [comment]
I think he means using the content dispostiton headers, so that we don't have to specify a file name manually.

I need something like this

**Comment by jeevatkm** [comment]
@RA341 Did you mean to get the filename from the content disposition instead of using SetOutput?

**Comment by RA341** [comment]
yeah

**Comment by jeevatkm** [comment]
I see. Currently, the SetOutput method does two things: it gets the filename input from the user and instructs the Resty to save the file.

Let me figure it out for the upcoming Resty v3.

**Comment by RA341** [comment]
this the snippet is use currently 



```go
resp, err := http.Get(downloadLink)
	if err != nil {
		return "", fmt.Errorf("failed to make request: %v", err)
	}
	if resp.StatusCode != 200 {
		return "", fmt.Errorf("invalid http code: %d, reason: %s", resp.StatusCode, resp.Status)
	}

	defer func(Body io.ReadCloser) {
		err := Body.Close()
		if err != nil {
			log.Fatal().Err(err).Msgf("failed to close response body")
		}
	}(resp.Body)

	// Get filename from Content-Disposition header
	_, params, err := mime.ParseMediaType(resp.Header.Get("Content-Disposition"))
	filename := ""
	if err == nil && params["filename"] != "" {
		filename = params["filename"]
	} else {
		// Fallback to URL path if header not available
		filename = path.Base(downloadLink)
	}

	if err := os.MkdirAll(downloadPath, 0775); err != nil {
		return "", fmt.Errorf("failed to create directories: %v", err)
	}

	destPath := fmt.Sprintf("%s/%s", downloadPath, filename)
	file, err := os.Create(destPath)
	if err != nil {
		return "", fmt.Errorf("failed to create file: %v", err)
	}
	defer func(file *os.File) {
		err := file.Close()
		if err != nil {
			log.Fatal().Err(err).Msgf("failed to close response body")
		}
	}(file)

	_, err = io.Copy(file, resp.Body)
	if err != nil {
		return "", fmt.Errorf("failed to copy response body: %v", err)
	}
```


no sure how much of its applicable

**Comment by jeevatkm** [comment]
@RA341 Thanks for sharing.

This part is applicable, the remaining lines are already present in the middleware.
```go
        // Get filename from Content-Disposition header
	_, params, err := mime.ParseMediaType(resp.Header.Get("Content-Disposition"))
	filename := ""
	if err == nil && params["filename"] != "" {
		filename = params["filename"]
	} else {
		// Fallback to URL path if header not available
		filename = path.Base(downloadLink)
	}
```

It needs to be blended with Resty Flow. Plus, a new method needs to be introduced beside the SetOutput.

**Comment by jeevatkm** [comment]
Implemented as part of PR #932 for Resty v3
